SonarQube.Client/Models/ServerSentEvents/SSEStreamReader.cs

```diff
@@ -92,19 +92,13 @@ private async Task<ISqServerEvent> ReadNextEventAsync()
         {
             try
             {
-                var sqEvent = await sqSSEStreamReader.ReadAsync();
-
-                return sqEvent;
-            }
-            catch (TaskCanceledException)
-            {
-                return null;
+                return await sqSSEStreamReader.ReadAsync();
             }
             catch (Exception ex)
             {
-                logger.Debug($"[SSEStreamReader] Failed to read sq event: {ex}");
-
-                return null;
+                logger.Warning($"[SSEStreamReader] Underlying stream failure while reading next event: {ex}");
+                sqSSEStreamReader.Dispose();
```

Comment:
I feel this is not done in the right place.
SSESession already disposes itself as a response to this message, so perhaps it should dispose `sseStreamReader`, that will dispose everything else. WDYT?

Reply:
I feel like, from a library design POV, this should happen here. Disposing it on SLVS side works only because we own both the library and the code that uses it. I think we should keep the knowledge about the internal components private. That doesn't mean we don't need to dispose it on SLVS side, it just disposes different things: one cancels the token in case of both regular and exceptional cases, the other disposes the network stream only in the case of exceptional failures. Plus, our Dispose methods are idempotent, so nothing bad is going to happen.

Reply:
`SSEStreamReader` isn't currently disposable, which is perhaps a bit odd.
However, that means at the moment this is the only place we can dispose this class.
I would leave it as it is for now and move on.

Reply:
Yes, it's leftovers from when reader and writer were separate and only the writer was disposable. When I changed the reader to hold the writer, I missed the dispose 🙈